### DISCUSSION TITLE
Object destructuring causes undefined (`interaction.options`)

### DISCUSSION THREAD

**Discussion by IsmaelTrentin** [discordjs/discord.js#8865] [Q&A] [answered]
I recently stumbled upon this weird behaviour of the get methods for `interaction.options`:

When getting the `getInteger` function with object destructuring out of `interaction.options` 

```ts
const { getInteger } = interaction.options;
const id = getInteger('optname');
```

it causes this error:
```
TypeError: Cannot read properties of undefined (reading '_getTypedOption')
    at getInteger (E:\[...]\node_modules\discord.js\src\structures\CommandInteractionOptionResolver.js:171:25)
```

I tried with `getBoolean`, `getString`, `getInteger` and `getNumber`, the behaviour is the same in all of them.

Now i'm asking myself if i'm missing something, i'm just bad at TS/JS or if this is actually a bug.
Is the approach of destructuring wrong? If so why?

**Comment by MrMythicalYT** [comment, marked as answer]
This would be due to you unbinding the `this`, when destructuring. As you can see here, it tries to access `this` interally
https://github.com/discordjs/discord.js/blob/d2754802cc3479e6288cdbfcd48c76bf57e09a07/packages/discord.js/src/structures/CommandInteractionOptionResolver.js#L171

Destructuring these methods is not a good idea. If you really wanted to, make sure you [bind](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind) it to the `interaction.options` object again.

**Reply by IsmaelTrentin** [reply]
Ohh didn't know about that. Learned something new about JS.
Thank you for the clean answer and the bind example 👍
